packages/ec-core/src/operator/selector/lexicase.rs

```diff
@@ -95,3 +100,200 @@ where
             .context("The pool of candidates was empty")
     }
 }
+
+#[cfg(test)]
+#[rustversion::attr(before(1.81), allow(clippy::unwrap_used))]
+#[rustversion::attr(
+    since(1.81),
+    expect(
+        clippy::unwrap_used,
+        reason = "Panicking is the best way to deal with errors in unit tests"
+    )
+)]
+mod tests {
+    use std::iter::once;
+
+    use proptest::{
+        collection,
+        prelude::{any, Strategy},
+        prop_assert, prop_assert_eq,
+    };
+    use test_strategy::proptest;
+
+    use super::*;
+    use crate::individual::ec::EcIndividual;
+
+    // Generate a population from a vector of single scores, one per individual.
+    // These have to be converted to iterators using `once` so they're "like"
+    // arrays of scores.
+    fn population_from_single_scores<I>(scores: I) -> Vec<EcIndividual<usize, TestResults<i32>>>
+    where
+        I: IntoIterator<Item = i32>,
+    {
+        // Mapping `once` here converts each single `i32` value into
+        // an iterator over `i32`s.
+        population_from_scores(scores.into_iter().map(once))
+    }
+
+    // Generate a population from a collection of score collections (e.g., arrays),
+    // one per individual.
+    fn population_from_scores<I, J>(scores: I) -> Vec<EcIndividual<usize, TestResults<i32>>>
+    where
+        I: IntoIterator<Item = J>,
+        J: IntoIterator<Item = i32>,
+    {
+        scores
+            .into_iter()
+            .map(TestResults::<i32>::from)
+            // We'll use the index as the "genome".
+            .enumerate()
+            .map(EcIndividual::from)
+            .collect::<Vec<_>>()
+    }
+
+    #[test]
+    fn single_best_single_error() {
+        // 9 is the sole winner in this population
+        let population = population_from_single_scores([5, 8, 9, 6, 3, 2, 0]);
+
+        let lexicase = Lexicase::new(1);
+        let mut rng = rand::thread_rng();
+
+        assert_eq!(&2, lexicase.select(&population, &mut rng).unwrap().genome());
+    }
+
+    #[test]
+    fn multiple_copies_best() {
+        // The two 9s are the possible winners
+        let population = population_from_single_scores([5, 8, 9, 6, 3, 2, 0, 9]);
+
+        let lexicase = Lexicase::new(1);
+        let mut rng = rand::thread_rng();
+
+        let selected = *lexicase.select(&population, &mut rng).unwrap().genome();
+        assert!(
+            selected == 2usize || selected == 7usize,
```

Comment:
```suggestion
            [2usize,7usize].contains(selected)
```

this is probably easier to maintain - if you prefer the way you've done it here this is totally fine by me as well tho.